pom.xml

```diff
@@ -749,6 +755,7 @@
                 <id>attach-sources</id>
                 <goals>
                   <goal>jar</goal>
+                  <goal>test-jar</goal> <!-- TODO ideally would be skipped unless we have a test-jar execution of maven-jar-plugin -->
```

Comment:
Tested against `workflow-step-api` (correctly includes the test sources); `workflow-api` (includes test sources even though there is no test JAR; unfortunately but mostly harmless); `pipeline-github-lib` (no `src/test/` so the source plugin automatically skips it).